### PR TITLE
Story 0 b cohort cards

### PR DESCRIPTION
**Pull request by hushangni**
Ticket :
https://trello.com/c/hRAhpYam/9-0-b-as-an-admin-i-would-like-a-page-where-i-can-see-all-of-my-previous-cohort-applications-listed-so-that-i-can-view-the-form-fo

Notes:
* added action
* added to reducer for cohorts
* added useEffect call to our backend in adminDashboard.js

QA:
* check that data pulled from firebase matches the data in firebase
* check that if data is loading, a loading state shown on screen

### CODE REVIEW

**Comment by e-schultz** [comment]
Hi,

I shared some notes on this in the slack channel, but attaching to the PR for historical sake ;)

The source of the issue is an eslint rule failing on the admin dashboard:

Screen Shot 2019-09-09 at 12 55 11 PM

Interestingly, when running 'npm run build' locally - the eslint rule won't fail the build, and it would be fine - I went on a bit of a search on some other issues around making the rule pass anyways, but I still had a nagging 'why does build work locally but fail on CI?'

I then checked the logs again, and noticed this:

```
Treating warnings as errors because process.env.CI = true.
Most CI servers set it automatically.
```

So even when doing a "production" build locally - warnings are warnings and not treated as error, but when on CI - a warning will get treated as an error. (something new I learned today)

testing it out locally on my machine (command may vary depending on the operating system and shell)

```bash
export CI=true
npm run build
```

and ... I then get the error,  good to know.

Most of below is from the slack chat:

for the useEffect hook - we need to specify what data/dependencies it has, 

if there is no array - it will run every render
if there is an empty array - it will run only once
if there is an array with values - it will only run when those change

I haven't played around as much with mixing react-redux connected components and hooks yet, and know that the latest release of react-redux does have some hooks

Without going too far down a rabbit-hole, we could possibly:

- pull the values off of props that we care about, and pass those in as a dependency, for example:

```jsx
const { getCohorts } = props;
  useEffect(() => {
    getCohorts();
  }, [getCohorts]);
 ```

 Simply add props instead:

 ```jsx
 useEffect(() => {
      const { getCohorts } = props;
      getCohorts();
  }, [props])
``` 

The potential issue with this is if anything in the props change - even if it's not related to why we want to fetch data, it could trigger another fetch.

Or, we could disable the eslint rule for that line to prevent the build from failing:


```jsx
const AdminDashboard = (props) => {
  useEffect(() => {
      const { getCohorts } = props;
      getCohorts();
  // we only want to fetch getCohorts when the Dashboard first loads,
  // so passing an empty array - we might want to revisit this later
  // and see if there is a better approach than disabling the rule
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
```

The eslint rules are generally pretty good, but occasionally we might need to disable them for a specific line or file.

If you wanted to disable a rule for an entire file, could add something like below to the top:

```
/* eslint-disable react-hooks/exhaustive-deps */
```

Generally, unless I have a good reason - I don't disable a rule for an entire file. 
But, if if there is a need - doing it for the next line.

I'd also tend to put a reason to why I'm disabling it - often it's better to try and fix the rule that is breaking than just disabling, and we don't want it to become a crutch every time we run into an eslint issue. 

The eslint documentation has more details on disabling rules with comments: https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments

tl;dr ---- we know that we only want to run the fetch when the screen first loads (for now, this may change) - I might lean towards disabling the rule for the one line, and revisiting it later on.
